### ISSUE TITLE
Add Pushover alerting

### ISSUE THREAD

**Issue by derekoharrow** [TwiN/gatus#129]
Please can you add Pushover alerting to gatus (which is great BTW!)

Details on how to send pushover alerts can be found here - https://pushover.net/api

I have managed to get this working using the custom alert, but it's quite clunky and requires encoding the user key, api key, message, etc. in the custom alert body text.

Thanks

**Comment by TwinProduction** [comment]
Could you please provide an example of how you got it working with the `custom` provider? 
It should help for the implementation.

I can't promise I'll be able to work on this in the short term, but if anybody wants to give it a shot, be my guest.

**Comment by derekoharrow** [comment]
> Could you please provide an example of how you got it working with the `custom` provider?
> It should help for the implementation.
> 
> I can't promise I'll be able to work on this in the short term, but if anybody wants to give it a shot, be my guest.

Sure - see below...

```
alerting:
  custom:
    url: https://api.pushover.net/1/messages.json
    method: "POST"
    body: |
      token=XXXXXXXXX&user=YYYYYYY&title=[ALERT_TRIGGERED_OR_RESOLVED]:+[SERVICE_NAME]&message=[ALERT_DESCRIPTION]
    default-alert:
      enabled: true
      description: "Healthcheck failed"
      failure-threshold: 3
      success-threshold: 3
      send-on-resolved: true
```

Other Pushover API parameters can also be passed in the HTML body, and it would be great to be able to override some of these for each service - especially Priority:

- attachment - an image attachment to send with the message; see attachments for more information on how to upload files
- device - your user's device name to send the message directly to that device, rather than all of the user's devices (multiple devices may be separated by a comma)
- title - your message's title, otherwise your app's name is used
- url - a supplementary URL to show with your message
- url_title - a title for your supplementary URL, otherwise just the URL is shown
- priority - send as -2 to generate no notification/alert, -1 to always send as a quiet notification, 1 to display as high-priority and bypass the user's quiet hours, or 2 to also require confirmation from the user
- sound - the name of one of the sounds supported by device clients to override the user's default sound choice
- timestamp - a Unix timestamp of your message's date and time to display to the user, rather than the time your message is received by our API

**Comment by TwiN** [comment]
Closed by #405
